### PR TITLE
Run cargo clippy as part of the CI

### PR DESCRIPTION
**Pull request by alistair23**

### CODE REVIEW

File: src/io/bit_buf.rs [hunk 1]
```diff
@@ -119,6 +124,12 @@ impl BitBuf {
     }
 }
 
+impl Default for io::bit_buf::BitBuf {
```
Comment (mcy) [inline src/io/bit_buf.rs:127]: Shouldn't this be just `BitBuf`?

**Review by mcy — APPROVED** [review]
Was meaning to get to this for a while. Thanks Alistair!

File: src/io/bit_buf.rs (hunk 1 above)
Comment (alistair23) [reply src/io/bit_buf.rs]: Fixed.

**Comment by alistair23** [comment]
Ping!

**Review by mcy — APPROVED** [review]
LG. =)

**Pull request merged**